### ISSUE TITLE
Debug mode?

### ISSUE THREAD

**Issue by JackYoungblood** [nojhan/liquidprompt#610]
Shell: bash
Operating system: Lubuntu 18.04 
Liquid Prompt version (1.11, 488bb5975db9fa8be0b795019f0b6d99ec792f50): 

Liquid Prompt is a great utility and I'm enjoying playing with it. There are so many flags that can be set that I'm having a hard  time determining which flags contributed to the final console prompt I see. It would be great if the tool had some kind of debug mode to explain which flags contributed to the final console prompt that I see. 

Right now my prompt says:
10:54:27 θ127° [:~/install] $ 

I can't figure out what the θ127° represents and which flags caused those values to be added to the output. I don't see the theta symbol, θ, in the README file.

**Comment by Rycieos** [comment]
The `θ` symbol is the default symbol for temperature:
https://github.com/nojhan/liquidprompt/blob/af63c4cfb8afe3f3161f3ace4aba3b940847a7a8/liquidprompt#L346

I really hope there is a bug in your system, because a temp of 127° seems a bit high. Try running either `sensors -u` or `acpi -t` to see what Liquidprompt is looking at to see your temps. If one of those is correctly reporting that temp, then there isn't a bug in Liquidprompt.

The request for a debug mode is a good one, and something we should look into. I'll keep this open for that.

In the meantime, if you want to see a crazy amount of debug info, you can run `set -x`, and you will see everything Liquidprompt is doing to build your PS1. Run `set +x` to disable.

**Comment by JackYoungblood** [comment]
Thank you for the response. Note that my machine does not feel particularly warm. The laptop is an HP model. A web search showed a few other ppl with a problem of a high reported temp on  acpitz-virtual-0 temp6. Should this be a cause for concern?

Here's the output from my `sensors -u`:
```
coretemp-isa-0000
Adapter: ISA adapter
Package id 0:
  temp1_input: 68.000
  temp1_max: 87.000
  temp1_crit: 105.000
  temp1_crit_alarm: 0.000
Core 0:
  temp2_input: 68.000
  temp2_max: 87.000
  temp2_crit: 105.000
  temp2_crit_alarm: 0.000
Core 1:
  temp3_input: 60.000
  temp3_max: 87.000
  temp3_crit: 105.000
  temp3_crit_alarm: 0.000
Core 2:
  temp4_input: 58.000
  temp4_max: 87.000
  temp4_crit: 105.000
  temp4_crit_alarm: 0.000
Core 3:
  temp5_input: 60.000
  temp5_max: 87.000
  temp5_crit: 105.000
  temp5_crit_alarm: 0.000

acpitz-virtual-0
Adapter: Virtual device
temp1:
  temp1_input: 59.000
  temp1_crit: 128.000
temp2:
  temp2_input: 54.000
  temp2_crit: 128.000
temp3:
  temp3_input: 47.000
  temp3_crit: 128.000
temp4:
  temp4_input: 47.000
  temp4_crit: 128.000
temp5:
  temp5_input: 32.000
  temp5_crit: 128.000
temp6:
  temp6_input: 127.000
  temp6_crit: 128.000

radeon-pci-0100
Adapter: PCI adapter
temp1:
  temp1_input: 54.000
  temp1_crit: 120.000
  temp1_crit_hyst: 90.000

```

**Comment by Rycieos** [comment]
> ```
> temp6:
>   temp6_input: 127.000
> ```
It appears that you do have a sensor reporting that high. My guess is that the sensor is faulty. You can try this tip [from this comment](https://github.com/nojhan/liquidprompt/issues/445#issuecomment-414130584) to disable that specific sensor output:

Write a file `/etc/sensors.d/iwlwifi` with the following contents:
```
chip "acpitz-virtual-0"
    ignore temp6
```

**Comment by JackYoungblood** [comment]
Thank you for the suggestion, especially considering that this is outside
the scope of the liquid prompt utility.

**Comment by Rycieos** [comment]
Pushing this to v2.1 as v2.0 is already too large a changeset.

I'm not sure debug mode is realistic without hurting runtime speed. I'll keep this open for now until I can determine for sure.

**Comment by asmodehn** [comment]
Posting here since I have a similar problem, and I am looking for a way to debug it :

With liquidprompt at v2.0.4:
My machine is not particularly warm, but the prompt says otherwise `θ84° [alexv:~] $ `

Attempting to find where this is coming from:
```
θ84° [alexv:~] $ acpi -t
Thermal 0: ok, 46.0 degrees C

θ84° [alexv:~] $ sensors -u | grep input
  temp1_input: 45.000
  temp2_input: 44.000
  temp3_input: 44.000
  temp4_input: 43.000
  temp5_input: 43.000
  temp6_input: 45.000
  temp7_input: 42.000
  temp1_input: 60.000
  temp1_input: 24.850
  in0_input: 16.992
  curr1_input: 0.463
  temp1_input: 38.000
  fan1_input: 3648.000
  fan2_input: 3267.000
  temp1_input: 48.000
  temp2_input: 0.000
  temp1_input: 42.850
  temp2_input: 42.850
  temp3_input: 48.850
  temp1_input: 48.000
```

Any other place where I should look for a temperature that says 84 C ?

**Comment by Rycieos** [comment]
This one is my mistake. You found a bug; one that I already fixed on the `master` branch, but I somehow missed was also on the stable branch. This is why I need to wrap up this changeset and just release 2.1 already...

Anyway, duplicate of #686, which was caused by 4c81ef4026c3105a260ff2b7e275bc8e50921c3b and fixed in b6a7189c57d4f96fdc51c025225013ff9870d8d0. What I didn't remember at that time is that I had backported 4c81ef4026c3105a260ff2b7e275bc8e50921c3b as 2c91f9854da7cfa549dc0e3def05f02139b99c51 to [v2.0.3](https://github.com/nojhan/liquidprompt/releases/tag/v2.0.3), so the bug was still in the 2.0/stable line.

Simple fix; I can just backport b6a7189c57d4f96fdc51c025225013ff9870d8d0 to stable, and release v2.0.5.

Note this is off topic for this issue. I see why you thought it was related, but the original issue is a different request. I'm just solving this without opening a new issue since there is already an issue for this (#686), and I will be closing this issue shortly (see following comment).

**Comment by Rycieos** [comment]
I am closing this issue, because I do not think it would be simple or useful.

Both Bash and Zsh have debugging modes built in (`set -x`), which provide much more information than a custom solution could possibly provide. That is what I personally use to debug Liquidprompt, and any other custom solution would slow down prompt generation, which is not a sacrifice I am willing to make.
